### PR TITLE
Call nBallerina from jBallerina (v2.0.0-beta.1)

### PR DESCRIPTION
**Pull request by ushirask**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/nballerina/nBallerinaCaller.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    private nBallerinaCaller(CompilerContext context) {
+        context.put(NBALLERINA, this);
+        CompilerOptions compilerOptions = CompilerOptions.getInstance(context);
+        this.nBal = compilerOptions.get(CompilerOptionName.NBALLERINA);
+    }
+
+    public  void callnBallerina(BLangPackage pkgNode)  {
+        if (nBal != null) {
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/nballerina/nBallerinaCaller.java:50]: convert this to early return style. ie: if null then return.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/nballerina/nBallerinaCaller.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                Function<Object[], Object> func = objects -> {
+                    try {
+                        return m.invoke(null, objects[0]);
+                    } catch (InvocationTargetException e) {
+                        Throwable targetException = e.getTargetException();
+                        if (targetException instanceof RuntimeException) {
+                            throw (RuntimeException) targetException;
+                        } else {
+                            throw new RuntimeException(targetException);
+                        }
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/nballerina/nBallerinaCaller.java:67]: If we use `throw new BallerinaException("Error invoking nBallerina backend", e);` will it print correct cause ?
if so we can use that, if not we can still throw `BallerinaException` don't think `RuntimeException` is the correct thing to throw from here.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/nballerina/nBallerinaCaller.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                final Throwable t = out.panic;
+                if (t != null) {
+                    if (t instanceof io.ballerina.runtime.internal.util.exceptions.BLangRuntimeException) {
+                        throw new BLangRuntimeException(t.getMessage());
+                    }
+                    if (t instanceof ErrorValue) {
+                        throw new BLangRuntimeException(
+                                "error: " + ((ErrorValue) t).getPrintableStackTrace());
+                    }
+                    throw (RuntimeException) t;
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/nballerina/nBallerinaCaller.java:94]: same as above.

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java [hunk 4]
```diff
@@ -305,6 +308,7 @@ private BuildOptions constructBuildOptions() {
                 .cloud(cloud)
                 .dumpBir(dumpBIR)
                 .dumpBirFile(dumpBIRFile)
+                .nBal(nBal)
```
Comment (manuranga) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java:311]: don't think phrase `nBal` matches with rest of the java code, were we have use longer descriptive names.

File: compiler/ballerina-lang/spotbugs-exclude.xml [hunk 5]
```diff
@@ -325,4 +325,7 @@
         <Class name="io.ballerina.projects.CompilerPluginManager"/>
         <Bug pattern="BC_UNCONFIRMED_CAST"/>
     </Match>
+    <Match>
+        <Package name="org.wso2.ballerinalang.compiler.nballerina"/>
```
Comment (manuranga) [inline compiler/ballerina-lang/spotbugs-exclude.xml:329]: what are the spotbugs you are getting ?

Comment (ushirask) [reply compiler/ballerina-lang/spotbugs-exclude.xml:329]: It doesn't allow to create a classloader, I'll try using a doPrivileged block instead

Comment (manuranga) [reply compiler/ballerina-lang/spotbugs-exclude.xml:329]: if creating the classloader is the only issue, then it's ok. can you paste exact error please.

Comment (ushirask) [reply compiler/ballerina-lang/spotbugs-exclude.xml]: I manged to fix it using a doPrivileged block, we no longer need to exclude the file

**Pull request closed without merging**